### ISSUE TITLE
半精度 _decay_step_counter() 报错

### ISSUE THREAD

**Issue by wickai** [PaddlePaddle/Paddle#16465]
出错位置在：fluid.layers.learning_rate_scheduler._decay_step_counter() 
半精度相关代码参照paddlecv的训练代码，配置fp16：https://github.com/PaddlePaddle/models/tree/develop/fluid/PaddleCV/image_classification

- 版本、环境信息：
    1）PaddlePaddle版本：1.3
    3）GPU：CUDA8
    4）系统环境：centos 
- 训练信息
    3）Operator信息
![image](https://user-images.githubusercontent.com/14268278/55000176-e860d300-500c-11e9-9d2a-bb92200fcea2.png)

**Comment by SunGaofeng** [comment]
从报错信息看，是参数名称冲突了，已经创建了一个名称为cast_0.tmp_0的参数，类型为FP16；后来又创建了一个同名的参数，但类型为FP32。
能贴下源代码吗，看下有哪些地方创建了cast op？

**Comment by wickai** [comment]
> 从报错信息看，是参数名称冲突了，已经创建了一个名称为cast_0.tmp_0的参数，类型为FP16；后来又创建了一个同名的参数，但类型为FP32。
> 能贴下源代码吗，看下有哪些地方创建了cast op？

           queue_capacity = 64
            py_reader = fluid.layers.py_reader(
                capacity=queue_capacity,
                shapes=[[-1] + image_shape, [-1, 1]],
                lod_levels=[0, 0],
                dtypes=[dtype, "int64"],
                use_double_buffer=True)
            image, label = fluid.layers.read_file(py_reader)
            if args.fp16:
                image = fluid.layers.cast(image, "float16")


另外直接用这里的转fp16代码： https://github.com/PaddlePaddle/models/blob/develop/fluid/PaddleCV/image_classification/utils/fp16_utils.py
